prefect_kubernetes/jobs.py

```diff
@@ -299,3 +309,208 @@ def kubernetes_orchestrator():
+                        name=pod_name,
+                        container=latest_v1_job.spec.template.spec.containers[0].name,
+                        **self._kubernetes_job.api_kwargs,
+                    )
+
+                if latest_v1_job.status.active:
+                    await sleep(self._kubernetes_job.interval_seconds)
+                    if self._kubernetes_job.timeout_seconds:
+                        elapsed_time += self._kubernetes_job.interval_seconds
+                elif latest_v1_job.status.failed:
+                    raise RuntimeError(
+                        f"Job {latest_v1_job.metadata.name!r} failed, check the "
+                        "Kubernetes pod logs for more information."
+                    )
+                elif latest_v1_job.status.succeeded:
+                    self._completed = True
+                    self.logger.info(
+                        f"Job {latest_v1_job.metadata.name!r} has completed."
+                    )
+
+        if self._kubernetes_job.delete_after_completion:
+            await self._cleanup()
+
+    async def fetch_result(self) -> Dict[str, Any]:
+        """Fetch the results of the job.
+
+        Returns:
+            The logs from each of the pods in the job.
+
+        Raises:
+            ValueError: If this method is called when the job has
+                a non-terminal state.
+        """
+
+        if not self._completed:
+            raise ValueError(
+                "The Kubernetes Job run is not in a completed state - "
+                "be sure to call `wait_for_completion` before attempting "
+                "to fetch the result."
+            )
+        return self.pod_logs
+
+
+class KubernetesJob(JobBlock):
+    """A block representing a Kubernetes job configuration."""
+
+    v1_job: Dict[str, Any] = Field(
+        default=...,
+        description=(
+            "The Kubernetes job manifest to run. This dictionary can be produced "
+            "using `yaml.safe_load`."
+        ),
+    )
+    api_kwargs: Dict[str, Any] = Field(
+        default_factory=dict,
+        description="The kwargs to pass to all Kubernetes API calls.",
+        example={"pretty": "true"},
+    )
+    credentials: KubernetesCredentials = Field(
+        default=..., description="The credentials to configure a client from."
+    )
+    delete_after_completion: bool = Field(
+        default=True,
+        description="Whether to delete the job after it has completed.",
+    )
+    interval_seconds: int = Field(
+        default=5,
+        description="The number of seconds to wait between job status checks.",
+    )
+    namespace: str = Field(
+        default="default",
+        description="The namespace to create and run the job in.",
+    )
+    timeout_seconds: Optional[int] = Field(
+        default=None,
+        description="The number of seconds to wait for the job run before timing out.",
+    )
+
+    _block_type_name = "Kubernetes Job"
```

Comment:
We need an alternate block type slug to avoid colliding with the block in the core library.
```suggestion
    _block_type_name = "Kubernetes Job"
    _block_type_slug = "k8s-job"
```